packages/mobile/src/sentry/Sentry.ts

```diff
@@ -1,32 +1,66 @@
 import * as Sentry from '@sentry/react-native'
 import DeviceInfo from 'react-native-device-info'
 import { select } from 'redux-saga/effects'
-import { SENTRY_CLIENT_URL } from 'src/config'
+import { sentryTracesSampleRateSelector } from 'src/app/selectors'
+import {
+  BLOCKSCOUT_BASE_URL,
+  DEFAULT_FORNO_URL,
+  EXCHANGE_PROVIDER_LINKS,
+  SENTRY_CLIENT_URL,
+  SPEND_MERCHANT_LINKS,
+} from 'src/config'
+import networkConfig from 'src/geth/networkConfig'
 import Logger from 'src/utils/Logger'
 import { currentAccountSelector } from 'src/web3/selectors'
 
 const TAG = 'sentry/Sentry'
 
 // Set this to true, if you want to test Sentry on dev builds
 export const SENTRY_ENABLED = !__DEV__ || false
+export const sentryRoutingInstrumentation = new Sentry.ReactNavigationInstrumentation()
 
-if (SENTRY_ENABLED) {
-  installSentry()
-} else {
-  Logger.info(TAG, 'Sentry not enabled')
-}
+export function* initializeSentry() {
+  if (!SENTRY_ENABLED) {
+    Logger.info(TAG, 'Sentry not enabled')
+    return
+  }
 
-// This should be called as early in the lifecycle of the app as possible.
-function installSentry() {
   if (!SENTRY_CLIENT_URL) {
     Logger.info(TAG, 'installSentry', 'Sentry URL not found, skiping instalation')
     return
   }
+
+  const tracesSampleRate = yield select(sentryTracesSampleRateSelector)
+  const tracingOrigins = [
```

Comment:
for the services we control, this makes sense.  by why add the tracing header to the 3rd party services?

https://docs.sentry.io/platforms/react-native/performance/instrumentation/automatic-instrumentation/#tracingorigins

Reply:
we don't have control over the performance of 3rd party services but if they are causing our app performance to drop, it is nice to know this upfront so that we don't waste time investigating the cause? i thought the extra visibility was nice, but i don't have strong feelings about this so i can remove if you think we should not track

Reply:
(but yes i forgot to mark this PR as WIP before - the list of tracingOrigins is something I still need to refine)

Reply:
🤔  i tried to go through the urls that we have in config to see what they do/if they should be included, but i'm not sure if i've understood everything correctly. i've excluded third party urls that are being used to generate webviews that we don't want to measure, but maybe there are still things in there we don't want to include as part of our traces?

Reply:
My expectation for this feature is to provide inter-service traces that help pinpoint failures and bottlenecks across all our services.  e.g., you can follow a request from the wallet -> blockchain API -> some other service.  that feature requires all services in the trace to be using sentry and our account.  So it makes sense to configure that information for services we own (and might use sentry today, or will sometime in the future).  But it's unclear why we'd do if for services we don't own (and won't ever run sentry with our account).

Reply:
yeah you're right, i guess i'm not 100% sure about which services you consider we "own"? the thing i was trying to achieve with these tracingOrigins is to provide more detail to the out of the box navigation tracing that Sentry provides - without these tracing urls, we would only see the "navigation" trace without any other information. i went through again in more detail and removed any urls that do not directly influence the screen navigation events. of course there are lots of other things we could be tracking to give this navigation trace more detail (like react profiling and custom instrumentation) but i've stuck to the minimum for now...the react profiling is very intrusive, we need to export every component we want to track wrapped in `Sentry.withProfiler` 🙈

<img width="1126" alt="Screenshot 2021-12-20 at 12 10 35" src="https://user-images.githubusercontent.com/20150449/146776676-2e6b2bea-70ad-48f0-8970-17256b511f22.png">

Reply:
I'm not too sure what's the recommended way to do this with Sentry, but even if we don't own a service (ODIS, segment, firebase, etc), we'd still want to know how long it takes to request from them.
Reading https://docs.sentry.io/product/sentry-basics/tracing/distributed-tracing/ it seems adding them as origins would be the way to go.
What do you think @kathaypacific @silasbw?

Reply:
This is also my original thought, tracing all services we use so that we can see their impact on our own trace events.

I ended up removing most of them because I couldn't verify that the traces were coming through correctly - either they did not contribute to the traces currently being traced (i.e. the ui load and navigation) or the services are not allowing the sentry trace header to be added to the requests. I inspected the network requests going through flipper on screen navigation and checked the corresponding events in Sentry.

To make the trace work, the docs say `You will need to configure your web server CORS to allow the sentry-trace header` so that got me doubtful about leaving url's in the `tracingOrigins` when they might not be working. I thought it would be better to add to the `tracingOrigins` if we expand our use of Sentry performance monitoring?

Reply:
We're talking about two things:
* distributed tracing
* profiling

We should do profiling, which includes transactions times for services we don't own.

We should also do distributed tracing, but it's only going to work for services we own (and that also use sentry!).

Reply:
...i guess i have no idea how sentry tracing works or if tracing is coupled with distributed tracing (but if that's the case it's implied that traces can't include non-HTTP transactions, which doesn't seem very useful) :confused: 

@gnardini halp

Reply:
I'm confused by how all this works.

I agree in principle of the difference between profiling vs tracing. 

Sentry has two concepts for performance monitoring: A `Transaction` is the main thing which can have several `Span`s. HTTP requests should be spans within a transaction.

Sentry's React Native docs claim that they add all http requests as spans to the active transaction. However, they don't. When seeing the transaction details I see that HTTP requests show up as 'Breadcrumbs' which are a different thing not performance-related so they are detecting them but not adding them as spans. However, when we add the tracing origins thing they do show up as Spans. I tried googling around a bit but it doesn't look like there's a ton of information around this.

I'd say we can keep these things as tracing originas for now?